src/components/BaseForm.tsx

```diff
@@ -37,17 +37,65 @@ export default function BaseForm(props: BaseFormProps) {
 
   return (
     <div className="formWrapper" ref={formRef} data-testid="formWrapper">
-      <FormWithTheme
-        {...restProps}
-        focusOnFirstError={errorFocus}
-        className={`rjsf ${className ? className : ""}`}
-        validator={validator}
-        onSubmit={handleOnSubmit}
-        // @ts-ignore
-        fields={adminFields}
-        disabled={disabled || showSpinner || isSubmitting}
-      />
-      {(isSubmitting || showSpinner) && <Spinner />}
+      <ErrorBoundary>
+        <FormWithTheme
+          {...restProps}
+          focusOnFirstError={errorFocus}
+          className={`rjsf ${className ? className : ""}`}
+          validator={validator}
+          onSubmit={handleOnSubmit}
+          // @ts-ignore
+          fields={adminFields}
+          disabled={disabled || showSpinner || isSubmitting}
+        />
+        {(isSubmitting || showSpinner) && <Spinner />}
+      </ErrorBoundary>
     </div>
   );
 }
+
+/*
+  This is a wrapper for child components that could fail.
+  There is no way to do this as a functional component at this time (2024-05-03).
+  The official answer is to not have components that crash. (ok, fair)
+  But we have some highly configurable UI's with rjsf, so a user's typo could cause an issue.
+*/
+class ErrorBoundary extends Component {
+  declare state: {
+    thrown?: Error;
+  };
+
+  declare props: {
+    children: ReactNode;
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
+  static getDerivedStateFromError(thrown) {
```

Comment:
I see that this is the key method in this pattern https://react.dev/reference/react/Component#static-getderivedstatefromerror